src/components/shared/form_fields/TextValueField.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+	<div class="form-field" :class="[ 'form-field-text', { 'is-invalid': showError } ]">
+		<label :for="inputId" class="form-field-label">
+			{{ label }} <em v-if="labelHelpText">{{ labelHelpText }}</em>
+		</label>
+		<div v-if="helpText" class="form-field-help-text" :id="`${inputId}-help-text`">
+			{{ helpText }}
+		</div>
+		<div class="control text-form-input" :class="{ 'has-icons-right': showError, 'is-disabled': disabled }">
+			<input
+				ref="field"
+				:name="name"
+				:value="value"
+				class="input"
+				:id="inputId"
+				:class="{ 'is-danger': showError }"
+				type="text"
+				:autocomplete="autocomplete"
+				:autofocus="autofocus"
+				:placeholder="placeholder"
+				:disabled="disabled"
+				:required="required"
+				:aria-invalid="showError"
+				:aria-describedby="ariaDescribedby"
+				@blur="$emit('field-changed', name )"
+				@input="onInput"
+				@paste="onInput"
+			/>
+		</div>
+		<span v-if="showError" class="help is-danger" :id="`${inputId}-error`">{{ errorMessage }}</span>
+		<span class="field-info-message">
+			<slot name="message"/>
+		</span>
+	</div>
+</template>
+
+<script setup lang="ts">
+
+// This field should be used when you need to format the value as the donor inputs.
+// We can't use the normal TextField for this because formatting the model value
+// causes an event loop so instead we use the native HTML input value. It also
+// contains functionality for externally updating the cursor position, because
+// when we change the value of an input field the browser jumps the cursor to
+// the end. We need to reset it to the proper position to allow the donor to
+// edit the field properly. It's up to the external formatter to decided that
+// position.
```

Comment:
Thank you for this comment!

Since this component probably won't be used outside of the `DirectDebitField` component and is highly specialized, I'd like to suggest to rename it. Either with a name that hints at its general purpose (where it *might* be reused), e.g. `ReformattableTextField`, `TextFieldWithDetachedValueAndValueDisplay`. Or with a name related to DirectDebit, e.g. `DirectDebitInputField`

Reply:
What TextValueField 🤫